apps/wing/src/cli.ts

```diff
@@ -75,9 +81,15 @@ async function main() {
     .argument("<entrypoint...>", "all entrypoints to test")
     .addOption(
       new Option("-t, --target <target>", "Target platform")
-        .choices(["tf-aws", "sim", "awscdk"])
+        .choices(["tf-aws", "tf-azure", "tf-gcp", "sim", "awscdk"])
         .default("sim")
     )
+    .hook("preAction", async (cmd) => {
+      const target = cmd.opts().target;
+      if (target != "sim") {
+        process.env.PROGRESS = "1";
+      }
+    })
```

Comment:
You can't add it to the root program object, because th `cmd.opts()` is able to see the parameters passed only to that specific command.

Reply:
Ah, gotcha. WDYT about refactoring the logic to a new function so the hooks can call it?

Reply:
I don't think that's actually possible for the same reason why we duplicated the target option inside both the `compile` and `test` commands. If we were to define a `target` option at a higher level, it would have been always usable also for commands where it made no sense to be able to set it.

Reply:
I don't think that's actually possible for the same reason why we duplicated the `target` option inside both the `compile` and `test` commands. If we were to define a `target` option at a higher level, it would have been always usable also for commands where it made no sense to be able to set it.

Reply:
> If we were to define a target option at a higher level, it would have been always usable also for commands where it made no sense to be able to set it.

I see your point, I would be okay with duplicating the option's definition (they can even have different descriptions) as long as the way we handle setting the env is consolidated by taking the function passed to the hook and extracting it.